site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -156,10 +156,13 @@ public function downloadCourseMaterialZip($dir_name, $path) {
      * @Route("/{_semester}/{_course}/course_materials/modify_permission")
      * @AccessControl(role="INSTRUCTOR")
      */
-    public function modifyCourseMaterialsFilePermission($filenames, $checked) {
+    public function modifyCourseMaterialsFilePermission($checked) {
         $data=$_POST['fn'];
-        if(count($data)==1){
-            $filename = $filenames;
+        if(is_string($data) || count($data)==1){
+            $data = [$data];
+        }
+    
+        foreach ($data as $filename){
             if (!isset($filename) ||
                 !isset($checked)) {
                 $this->core->redirect($this->core->buildCourseUrl(['course_materials']));
```

Comment:
Github won't let me comment on the exact line because you didn't modify it, but it looks like we are using "now" as the default time here if something goes wrong. Let's use something else to avoid accidental file release (e.g. end of time).

Reply:
changed, default if no timestamp is given or error is now 'end of time'